src/GridFieldActionsMenu.php

```diff
@@ -0,0 +1,277 @@
+<?php
+
+namespace Symbiote\GridFieldExtensions;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+use Silverstripe\Core\Convert;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridField_ColumnProvider;
+use SilverStripe\Forms\GridField\GridField_URLHandler;
+use SilverStripe\Forms\GridField\GridFieldDetailForm;
+use SilverStripe\Forms\TabSet;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\View\ArrayData;
+use SilverStripe\View\SSViewer;
+
+/**
+ * Provides an encapsulated list of actions to do with the current record
+ * such as edit, publish, unpublish, archive, et al.
+ */
+class GridFieldActionsMenu implements
+    GridField_ColumnProvider,
+    GridField_URLHandler
+{
+    /**
+     * Whether to include the first Root tab in the actions list.
+     * The first entry generally relates to the default action, which often
+     * is also enacted by simply clicking on the row itself - so may be hidden
+     * in this case, or shown if clicking the row performs no action.
+     *
+     * @var boolean
+     */
+    protected $showFirstTab = true;
+
+    /**
+     * The action items to show in the menu
+     *
+     * Example structure:
+     *
+     * <code>
+     * // Actions
+     * [
+     *     // Group
+     *     [
+     *         // Action
+     *         [
+     *             'Title' => 'Content',
+     *             'Link' => 'hrefme',
+     *             'Type' => 'arbitrary_name', // e.g. 'link' or 'versioning'
+     *         ],
+     *         ...
+     *     ],
+     *     // or a callable that generates the above Group level array format
+     *     function($gridField, $record, $typeName) ...
+     *     ...
+     * ]
+     * </code>
+     *
+     * @var array
+     */
+    protected $actions = [];
+
+    public function __construct($showFirstTab = true)
+    {
+        $this->setShowFirstTab($showFirstTab);
+        $this->actions = [
+            'link' => [$this, 'addRootTabActions'],
+            'versioning' => [$this, 'addVersionedActions']
+        ];
+    }
+
+    public function augmentColumns($gridField, &$columns)
+    {
+        if (!in_array('Actions', $columns)) {
+            $columns[] = 'Actions';
+        }
+    }
+
+    public function getColumnsHandled($gridField)
+    {
+        return ['Actions'];
+    }
+
+    /**
+     * Add each of the "Root" tabs to the actions for this component
+     * We expect that a tabbed list of fields will always have a singular root.
+     *
+     * @param DataObject $record
+     * @param string $typeName
+     *
+     * @return array
+     */
+    protected function addRootTabActions(GridField $gridField, DataObject $record, $typeName)
```

Comment:
Doesn't match phpdoc.

Reply:
Fixed. Comment on line that hasn't changed, so doesn't "auto-resolve".